>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("a", 1)

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("a", 0)
Traceback (most recent call last):
ValueError: a must be greater than 0

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("a", "a")
Traceback (most recent call last):
TypeError: a must be an integer

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator()
Traceback (most recent call last):
TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator(5, 5, 5)
Traceback (most recent call last):
TypeError: integer_validator() takes 3 positional arguments but 4 were given

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("a", -5)
Traceback (most recent call last):
ValueError: a must be greater than 0

>>> base_geometry = BaseGeometry()
>>> base_geometry.area()
Traceback (most recent call last):
Exception: area() is not implemented

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("a")
Traceback (most recent call last):
TypeError: integer_validator() missing 1 required positional argument: 'value'

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("5")
Traceback (most recent call last):
TypeError: integer_validator() missing 1 required positional argument: 'value'

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("a", "a", 5)
Traceback (most recent call last):
TypeError: integer_validator() takes 3 positional arguments but 4 were given

>>> base_geometry = BaseGeometry()
>>> base_geometry.integer_validator("5")
Traceback (most recent call last):
TypeError: integer_validator() missing 1 required positional argument: 'value'

>>> base_geometry.area(5)
Traceback (most recent call last):
Exception: area() is not implemented

>>> base_geometry.integer_validator("a", 1, 1)
Traceback (most recent call last):
TypeError: integer_validator() takes 3 positional arguments but 4 were given

>>> base_geometry.integer_validator("a", True)
Traceback (most recent call last):
TypeError: Test must be an integer
